src/App.jsx

```diff
@@ -119,11 +121,41 @@ function App() {
     [shareDBDoc]
   );
 
+  const deleteFile = useCallback(
+    (key) => {
+      const currentDocument = shareDBDoc.data;
+      const nextDocument = { ...currentDocument };
+      delete nextDocument[key];
+      shareDBDoc.submitOp(diff(currentDocument, nextDocument));
+    },
+    [shareDBDoc]
+  );
+
+
+  const createFile = useCallback(() => {
+    const newName = prompt('Enter new file name');
+    if (!newName) return;
+    const currentDocument = shareDBDoc.data;
+    const nextDocument = {
+      ...currentDocument,
+      [randomId()]: {
+        name: newName,
+        content: '',
+      },
+    };
+    shareDBDoc.submitOp(diff(currentDocument, nextDocument));
```

Comment:
Nice. We may want to consider opening the new file also, immediately after creating it. I think VSCode does this.